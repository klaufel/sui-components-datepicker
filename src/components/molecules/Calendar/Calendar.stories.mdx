import {Meta, Props, Story, Preview} from '@storybook/addon-docs/blocks'
import {text} from '@storybook/addon-knobs'
import {actions} from '@storybook/addon-actions'
import Calendar from '.'

<Meta title="Molecules/Calendar" component={Calendar} />

# Calendar

Let's define a story for our `Calendar` component:

<Preview>
  <Story name="default">
    <Calendar />
  </Story>
</Preview>

## Props

<Props of={Calendar} />

### With a default date

<Preview>
  <Story name="with a default date">
    <Calendar date="01/10/1994" />
  </Story>
  <Story name="with a default date">
    <Calendar date="01/10/1994" />
  </Story>
</Preview>
