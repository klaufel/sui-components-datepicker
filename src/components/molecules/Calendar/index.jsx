import React, {useState} from 'react'
import PropTypes from 'prop-types'
import './index.scss'
import Icon from '../../atoms/Icon'
import {
  getCalendarDays,
  getMonthName,
  getNextMonth,
  getPrevMonth,
  getDateFormat,
  WEEKDAYS_NAMES
} from './dates'

const Calendar = ({onDateChange}) => {
  const actualDate = new Date()
  const [dateSelected, setDateSelected] = useState({
    year: actualDate.getFullYear(),
    month: actualDate.getMonth() + 1
  })
  const daysInMonth = getCalendarDays(dateSelected.year, dateSelected.month)
  const [dateFull, setDateFull] = useState(
    getDateFormat(dateSelected.year, dateSelected.month)
  )

  const handleNextMonth = () => {
    setDateSelected(getNextMonth(dateSelected.year, dateSelected.month + 1))
  }

  const handlePrevMonth = () => {
    setDateSelected(getPrevMonth(dateSelected.year, dateSelected.month - 1))
  }

  const handleClickDay = date => {
    setDateFull(date)
    onDateChange && onDateChange(date)
  }

  return (
    <div className="Calendar">
      <div className="Calendar-topBar">
        <span
          className="Calendar-arrow Calendar-arrow--left"
          onClick={handlePrevMonth}
        >
          <Icon name="chevronLeft" />
        </span>
        <span className="Calendar-month">
          {getMonthName(dateSelected.month)} / {dateSelected.year}
        </span>
        <span
          className="Calendar-arrow Calendar-arrow--right"
          onClick={handleNextMonth}
        >
          <Icon name="chevronRight" />
        </span>
      </div>
      <div className="Calendar-days">
        {WEEKDAYS_NAMES.map(name => (
          <div key={name} className="Calendar-dayWeeks">
            {name}
          </div>
        ))}
        {daysInMonth.map(dates => {
          const {day, date, month} = dates
          return (
            <div
              key={date}
              title={date}
              className={`Calendar-day Calendar-day--${month} ${date ===
                dateFull && 'is-selected'}`}
              onClick={() => handleClickDay(date)}
            >
              {day}
            </div>
          )
        })}
      </div>
    </div>
  )
}

Calendar.propTypes = {
  /** Callback to return a selected date */
  onDateChange: PropTypes.func
}

export default Calendar
