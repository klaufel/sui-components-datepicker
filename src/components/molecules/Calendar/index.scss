@import '~@schibstedspain/sui-theme/lib/index';

$Calendar-day-width: 44px !default;
$Calendar-day-onHover-backgroundColor: #dedede !default;
$Calendar-day-isSelected-backgroundColor: #31417d !default;
$Calendar-day-isSelected-color: #ffffff !default;

$Calendar-width: $Calendar-day-width * 7 !default;

.Calendar {
  background: #fff;
  max-width: $Calendar-width;
  margin: 0 auto;
}

.Calendar-topBar {
  display: flex;
  width: 100%;
  padding-bottom: 12px;
}

.Calendar-month {
  font-size: 16px;
  line-height: 20px;
  font-weight: 700;
  padding: 4px 8px;
  flex: 1 auto;
  text-align: center;
}

.Calendar-arrow {
  color: #0000ff;
  font-weight: 700;
  cursor: pointer;
  padding: 2px 4px;
}

.Calendar-days {
  font-size: 14px;
  display: grid;
  position: relative;
  grid-template-columns: repeat(7, $Calendar-day-width);
  grid-template-rows: 1fr repeat(6, $Calendar-day-width);
  border-radius: 50%;
  align-items: center;
  justify-content: center;
}

.Calendar-dayWeeks {
  display: block;
  padding: 8px 0;
  font-size: 12px;
  text-align: center;

  &--weekend {
    font-weight: 700;
  }
}


.Calendar-day {
  text-align: center;
  cursor: pointer;
  width: 100%;
  height: 100%;
  display: flex;
  align-items: center;
  justify-content: center;
  border-radius: 50%;
  position: relative;

  &--prev,
  &--next {
    color: #a6a6a6;
  }

  &:hover {
    background: $Calendar-day-onHover-backgroundColor;
  }

  &.is-selected {
    color: $Calendar-day-isSelected-color;
    background: $Calendar-day-isSelected-backgroundColor;
  }
}
