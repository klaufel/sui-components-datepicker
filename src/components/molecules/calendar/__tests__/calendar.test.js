import React from 'react'
import {render} from '@testing-library/react'
import Calendar from '..'

describe('Calendar', () => {
  test('it works with a year and month', () => {
    const {container} = render(<Calendar year={1970} month={6} />)
    expect(container.firstChild).toMatchSnapshot()
  })

  test('it works with a default date', () => {
    const {container} = render(<Calendar date="01/10/1994" />)
    expect(container.firstChild).toMatchSnapshot()
  })

  test('it works with a default date and year/month selected', () => {
    const {container} = render(
      <Calendar date="31/01/2020" year={2020} month={2} />
    )
    expect(container.firstChild).toMatchSnapshot()
  })
})
