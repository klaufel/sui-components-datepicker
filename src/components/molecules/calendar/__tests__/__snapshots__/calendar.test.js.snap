// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calendar it works with a default date 1`] = `
<div
  class="Calendar"
>
  <div
    class="Calendar-topBar"
  >
    <span
      class="Calendar-arrow Calendar-arrow--left"
    >
      <svg
        class="Icon Icon--md"
        enable-background="new 0 0 24 24"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M15.5 20.4L7.1 12l8.4-8.4L16.9 5l-7 7 7 7z"
        />
      </svg>
    </span>
    <span
      class="Calendar-month"
    >
      <span
        class="Heading Heading--sm"
      >
        October
         
        1994
      </span>
    </span>
    <span
      class="Calendar-arrow Calendar-arrow--right"
    >
      <svg
        class="Icon Icon--md"
        enable-background="new 0 0 24 24"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M8.5 20.4l8.4-8.4-8.4-8.4L7.1 5l7 7-7 7z"
        />
      </svg>
    </span>
  </div>
  <div
    class="Calendar-days"
  >
    <div
      class="Calendar-dayWeeks Calendar-dayWeeks--weekend"
    >
      Sun
    </div>
    <div
      class="Calendar-dayWeeks"
    >
      Mon
    </div>
    <div
      class="Calendar-dayWeeks"
    >
      Tue
    </div>
    <div
      class="Calendar-dayWeeks"
    >
      Wed
    </div>
    <div
      class="Calendar-dayWeeks"
    >
      Thu
    </div>
    <div
      class="Calendar-dayWeeks"
    >
      Fri
    </div>
    <div
      class="Calendar-dayWeeks Calendar-dayWeeks--weekend"
    >
      Sat
    </div>
    <div
      class="Calendar-day Calendar-day--prev"
    >
      25
    </div>
    <div
      class="Calendar-day Calendar-day--prev"
    >
      26
    </div>
    <div
      class="Calendar-day Calendar-day--prev"
    >
      27
    </div>
    <div
      class="Calendar-day Calendar-day--prev"
    >
      28
    </div>
    <div
      class="Calendar-day Calendar-day--prev"
    >
      29
    </div>
    <div
      class="Calendar-day Calendar-day--prev"
    >
      30
    </div>
    <div
      class="Calendar-day Calendar-day--current is-selected"
    >
      1
    </div>
    <div
      class="Calendar-day Calendar-day--current"
    >
      2
    </div>
    <div
      class="Calendar-day Calendar-day--current"
    >
      3
    </div>
    <div
      class="Calendar-day Calendar-day--current"
    >
      4
    </div>
    <div
      class="Calendar-day Calendar-day--current"
    >
      5
    </div>
    <div
      class="Calendar-day Calendar-day--current"
    >
      6
    </div>
    <div
      class="Calendar-day Calendar-day--current"
    >
      7
    </div>
    <div
      class="Calendar-day Calendar-day--current"
    >
      8
    </div>
    <div
      class="Calendar-day Calendar-day--current"
    >
      9
    </div>
    <div
      class="Calendar-day Calendar-day--current"
    >
      10
    </div>
    <div
      class="Calendar-day Calendar-day--current"
    >
      11
    </div>
    <div
      class="Calendar-day Calendar-day--current"
    >
      12
    </div>
    <div
      class="Calendar-day Calendar-day--current"
    >
      13
    </div>
    <div
      class="Calendar-day Calendar-day--current"
    >
      14
    </div>
    <div
      class="Calendar-day Calendar-day--current"
    >
      15
    </div>
    <div
      class="Calendar-day Calendar-day--current"
    >
      16
    </div>
    <div
      class="Calendar-day Calendar-day--current"
    >
      17
    </div>
    <div
      class="Calendar-day Calendar-day--current"
    >
      18
    </div>
    <div
      class="Calendar-day Calendar-day--current"
    >
      19
    </div>
    <div
      class="Calendar-day Calendar-day--current"
    >
      20
    </div>
    <div
      class="Calendar-day Calendar-day--current"
    >
      21
    </div>
    <div
      class="Calendar-day Calendar-day--current"
    >
      22
    </div>
    <div
      class="Calendar-day Calendar-day--current"
    >
      23
    </div>
    <div
      class="Calendar-day Calendar-day--current"
    >
      24
    </div>
    <div
      class="Calendar-day Calendar-day--current"
    >
      25
    </div>
    <div
      class="Calendar-day Calendar-day--current"
    >
      26
    </div>
    <div
      class="Calendar-day Calendar-day--current"
    >
      27
    </div>
    <div
      class="Calendar-day Calendar-day--current"
    >
      28
    </div>
    <div
      class="Calendar-day Calendar-day--current"
    >
      29
    </div>
    <div
      class="Calendar-day Calendar-day--current"
    >
      30
    </div>
    <div
      class="Calendar-day Calendar-day--current"
    >
      31
    </div>
    <div
      class="Calendar-day Calendar-day--next"
    >
      1
    </div>
    <div
      class="Calendar-day Calendar-day--next"
    >
      2
    </div>
    <div
      class="Calendar-day Calendar-day--next"
    >
      3
    </div>
    <div
      class="Calendar-day Calendar-day--next"
    >
      4
    </div>
    <div
      class="Calendar-day Calendar-day--next"
    >
      5
    </div>
  </div>
</div>
`;

exports[`Calendar it works with a default date and year/month selected 1`] = `
<div
  class="Calendar"
>
  <div
    class="Calendar-topBar"
  >
    <span
      class="Calendar-arrow Calendar-arrow--left"
    >
      <svg
        class="Icon Icon--md"
        enable-background="new 0 0 24 24"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M15.5 20.4L7.1 12l8.4-8.4L16.9 5l-7 7 7 7z"
        />
      </svg>
    </span>
    <span
      class="Calendar-month"
    >
      <span
        class="Heading Heading--sm"
      >
        February
         
        2020
      </span>
    </span>
    <span
      class="Calendar-arrow Calendar-arrow--right"
    >
      <svg
        class="Icon Icon--md"
        enable-background="new 0 0 24 24"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M8.5 20.4l8.4-8.4-8.4-8.4L7.1 5l7 7-7 7z"
        />
      </svg>
    </span>
  </div>
  <div
    class="Calendar-days"
  >
    <div
      class="Calendar-dayWeeks Calendar-dayWeeks--weekend"
    >
      Sun
    </div>
    <div
      class="Calendar-dayWeeks"
    >
      Mon
    </div>
    <div
      class="Calendar-dayWeeks"
    >
      Tue
    </div>
    <div
      class="Calendar-dayWeeks"
    >
      Wed
    </div>
    <div
      class="Calendar-dayWeeks"
    >
      Thu
    </div>
    <div
      class="Calendar-dayWeeks"
    >
      Fri
    </div>
    <div
      class="Calendar-dayWeeks Calendar-dayWeeks--weekend"
    >
      Sat
    </div>
    <div
      class="Calendar-day Calendar-day--prev"
    >
      26
    </div>
    <div
      class="Calendar-day Calendar-day--prev"
    >
      27
    </div>
    <div
      class="Calendar-day Calendar-day--prev"
    >
      28
    </div>
    <div
      class="Calendar-day Calendar-day--prev"
    >
      29
    </div>
    <div
      class="Calendar-day Calendar-day--prev"
    >
      30
    </div>
    <div
      class="Calendar-day Calendar-day--prev is-selected"
    >
      31
    </div>
    <div
      class="Calendar-day Calendar-day--current"
    >
      1
    </div>
    <div
      class="Calendar-day Calendar-day--current"
    >
      2
    </div>
    <div
      class="Calendar-day Calendar-day--current"
    >
      3
    </div>
    <div
      class="Calendar-day Calendar-day--current"
    >
      4
    </div>
    <div
      class="Calendar-day Calendar-day--current"
    >
      5
    </div>
    <div
      class="Calendar-day Calendar-day--current"
    >
      6
    </div>
    <div
      class="Calendar-day Calendar-day--current"
    >
      7
    </div>
    <div
      class="Calendar-day Calendar-day--current"
    >
      8
    </div>
    <div
      class="Calendar-day Calendar-day--current"
    >
      9
    </div>
    <div
      class="Calendar-day Calendar-day--current"
    >
      10
    </div>
    <div
      class="Calendar-day Calendar-day--current"
    >
      11
    </div>
    <div
      class="Calendar-day Calendar-day--current"
    >
      12
    </div>
    <div
      class="Calendar-day Calendar-day--current"
    >
      13
    </div>
    <div
      class="Calendar-day Calendar-day--current"
    >
      14
    </div>
    <div
      class="Calendar-day Calendar-day--current"
    >
      15
    </div>
    <div
      class="Calendar-day Calendar-day--current"
    >
      16
    </div>
    <div
      class="Calendar-day Calendar-day--current"
    >
      17
    </div>
    <div
      class="Calendar-day Calendar-day--current"
    >
      18
    </div>
    <div
      class="Calendar-day Calendar-day--current"
    >
      19
    </div>
    <div
      class="Calendar-day Calendar-day--current"
    >
      20
    </div>
    <div
      class="Calendar-day Calendar-day--current"
    >
      21
    </div>
    <div
      class="Calendar-day Calendar-day--current"
    >
      22
    </div>
    <div
      class="Calendar-day Calendar-day--current"
    >
      23
    </div>
    <div
      class="Calendar-day Calendar-day--current"
    >
      24
    </div>
    <div
      class="Calendar-day Calendar-day--current"
    >
      25
    </div>
    <div
      class="Calendar-day Calendar-day--current"
    >
      26
    </div>
    <div
      class="Calendar-day Calendar-day--current"
    >
      27
    </div>
    <div
      class="Calendar-day Calendar-day--current"
    >
      28
    </div>
    <div
      class="Calendar-day Calendar-day--current"
    >
      29
    </div>
  </div>
</div>
`;

exports[`Calendar it works with a year and month 1`] = `
<div
  class="Calendar"
>
  <div
    class="Calendar-topBar"
  >
    <span
      class="Calendar-arrow Calendar-arrow--left"
    >
      <svg
        class="Icon Icon--md"
        enable-background="new 0 0 24 24"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M15.5 20.4L7.1 12l8.4-8.4L16.9 5l-7 7 7 7z"
        />
      </svg>
    </span>
    <span
      class="Calendar-month"
    >
      <span
        class="Heading Heading--sm"
      >
        June
         
        1970
      </span>
    </span>
    <span
      class="Calendar-arrow Calendar-arrow--right"
    >
      <svg
        class="Icon Icon--md"
        enable-background="new 0 0 24 24"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M8.5 20.4l8.4-8.4-8.4-8.4L7.1 5l7 7-7 7z"
        />
      </svg>
    </span>
  </div>
  <div
    class="Calendar-days"
  >
    <div
      class="Calendar-dayWeeks Calendar-dayWeeks--weekend"
    >
      Sun
    </div>
    <div
      class="Calendar-dayWeeks"
    >
      Mon
    </div>
    <div
      class="Calendar-dayWeeks"
    >
      Tue
    </div>
    <div
      class="Calendar-dayWeeks"
    >
      Wed
    </div>
    <div
      class="Calendar-dayWeeks"
    >
      Thu
    </div>
    <div
      class="Calendar-dayWeeks"
    >
      Fri
    </div>
    <div
      class="Calendar-dayWeeks Calendar-dayWeeks--weekend"
    >
      Sat
    </div>
    <div
      class="Calendar-day Calendar-day--prev"
    >
      31
    </div>
    <div
      class="Calendar-day Calendar-day--current"
    >
      1
    </div>
    <div
      class="Calendar-day Calendar-day--current"
    >
      2
    </div>
    <div
      class="Calendar-day Calendar-day--current"
    >
      3
    </div>
    <div
      class="Calendar-day Calendar-day--current"
    >
      4
    </div>
    <div
      class="Calendar-day Calendar-day--current"
    >
      5
    </div>
    <div
      class="Calendar-day Calendar-day--current"
    >
      6
    </div>
    <div
      class="Calendar-day Calendar-day--current"
    >
      7
    </div>
    <div
      class="Calendar-day Calendar-day--current"
    >
      8
    </div>
    <div
      class="Calendar-day Calendar-day--current"
    >
      9
    </div>
    <div
      class="Calendar-day Calendar-day--current"
    >
      10
    </div>
    <div
      class="Calendar-day Calendar-day--current"
    >
      11
    </div>
    <div
      class="Calendar-day Calendar-day--current"
    >
      12
    </div>
    <div
      class="Calendar-day Calendar-day--current"
    >
      13
    </div>
    <div
      class="Calendar-day Calendar-day--current"
    >
      14
    </div>
    <div
      class="Calendar-day Calendar-day--current"
    >
      15
    </div>
    <div
      class="Calendar-day Calendar-day--current"
    >
      16
    </div>
    <div
      class="Calendar-day Calendar-day--current"
    >
      17
    </div>
    <div
      class="Calendar-day Calendar-day--current"
    >
      18
    </div>
    <div
      class="Calendar-day Calendar-day--current"
    >
      19
    </div>
    <div
      class="Calendar-day Calendar-day--current"
    >
      20
    </div>
    <div
      class="Calendar-day Calendar-day--current"
    >
      21
    </div>
    <div
      class="Calendar-day Calendar-day--current"
    >
      22
    </div>
    <div
      class="Calendar-day Calendar-day--current"
    >
      23
    </div>
    <div
      class="Calendar-day Calendar-day--current"
    >
      24
    </div>
    <div
      class="Calendar-day Calendar-day--current"
    >
      25
    </div>
    <div
      class="Calendar-day Calendar-day--current"
    >
      26
    </div>
    <div
      class="Calendar-day Calendar-day--current"
    >
      27
    </div>
    <div
      class="Calendar-day Calendar-day--current"
    >
      28
    </div>
    <div
      class="Calendar-day Calendar-day--current"
    >
      29
    </div>
    <div
      class="Calendar-day Calendar-day--current"
    >
      30
    </div>
    <div
      class="Calendar-day Calendar-day--next"
    >
      1
    </div>
    <div
      class="Calendar-day Calendar-day--next"
    >
      2
    </div>
    <div
      class="Calendar-day Calendar-day--next"
    >
      3
    </div>
    <div
      class="Calendar-day Calendar-day--next"
    >
      4
    </div>
  </div>
</div>
`;
